Reaqtor/Samples/Remoting/Reaqtor.Remoting.Client.Library/Operators.cs

```diff
@@ -383,6 +383,36 @@ public static IAsyncReactiveQbservable<IList<TSource>> Buffer<TSource>(this IAsy
                     source.Expression));
         }
 
+        /// <summary>
+        /// Returns a query observable sequence that contains all elements, or a default
+        /// item if the source query observable sequence contains no element.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements in the source sequence.</typeparam>
+        /// <param name="source">Source sequence whose elements to propagate.</param>
+        /// <param name="defaultValue">
+        /// The default item to propagate when the source sequence emits nothing.
+        /// </param>
+        /// <returns>
+        /// Query observable sequence containing all elements from the source query observable
+        /// sequence, or a default item.
+        /// </returns>
+        [KnownResource(Remoting.Client.Constants.Observable.DefaultIfEmpty.DefaultValue)]
+        public static IAsyncReactiveQbservable<TSource> DefaultIfEmpty<TSource>(
+            this IAsyncReactiveQbservable<TSource> source,
+            TSource defaultValue)
+        {
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            return source.Provider.CreateQbservable<TSource>(
+                Expression.Call(
+                    ((MethodInfo)MethodBase.GetCurrentMethod()).MakeGenericMethod(typeof(TSource)),
+                    source.Expression,
+                    Expression.Constant(defaultValue)));
```

Comment:
This needs to explicitly specify typeof(TSource) as a parameter to Constant. Otherwise, for null, it'd default to object and the Call factory would throw on type checking failures.

Reply:
As a best practice, let's also check that other calls to Constant specify the exact type, e.g. ElementAt for its int index.